src/actions/index.js

```diff
@@ -458,10 +501,108 @@ function updateOutput(
         });
     }
 
+    // Clientside hook
+    if (!isNil(clientside_function) && !isEmpty(clientside_function)) {
+        let returnValue;
+        try {
+            returnValue = window[clientside_function.namespace][
+                clientside_function.function_name
+            ](
+                ...(has('inputs', payload)
+                    ? pluck('value', payload.inputs)
+                    : []),
+                ...(has('state', payload) ? pluck('value', payload.state) : [])
+            );
+        } catch (e) {
+            /* eslint-disable no-console */
+            console.error(
+                `The following error occurred while executing ${
+                    clientside_function.namespace
+                }.${clientside_function.function_name} ` +
+                    `in order to update component "${payload.output}" ⋁⋁⋁`
+            );
+            console.error(e);
+            /* eslint-enable no-console */
+
+            /*
+             * Update the request queue by treating a successful clientside
+             * like a failed serverside response (500 status code)
+             */
+
+            /* eslint-disable no-magic-numbers */
+            updateRequestQueue(true, 500);
+            /* eslint-enable no-magic-numbers */
+            return;
+        }
+
+        function updateOutput(outputIdAndProp, outputValue) {
+            const [outputId, outputProp] = outputIdAndProp.split('.');
+            const updatedProps = {
+                [outputProp]: outputValue,
+            };
+
+            /*
+             * Update the request queue by treating a successful clientside
+             * like a succesful serverside response (200 status code)
+             */
+            updateRequestQueue(false, STATUS.OK);
+
+            // Update the layout with the new result
+            dispatch(
+                updateProps({
+                    itempath: getState().paths[outputId],
+                    props: updatedProps,
+                    source: 'response',
+                })
+            );
+
+            /*
+             * This output could itself be a serverside or clientside input
+             * to another function
+             */
+            dispatch(
+                notifyObservers({
+                    id: outputId,
+                    props: updatedProps,
+                })
+            );
+        }
+
+        if (payload.output.startsWith('..')) {
+            /*
+             * If this update is for multiple outputs, then it has
+             * starting & trailing `..` and each propId pair is separated
+             * by `...`, e.g.
+             * "..output-1.value...output-2.value...output-3.value...output-4.value.."
+             */
+            const outputPropIds = payload.output
+                .split('...')
+                .map(o => o.replace('..', ''));
```

Comment:
There's now 3 implementations of this multiple-outputs splitting with slight differences., This one, https://github.com/plotly/dash-renderer/blob/master/src/actions/index.js#L247, and https://github.com/plotly/dash-renderer/blob/master/src/reducers/dependencyGraph.js#L26. 🌴

Reply:
🙈